src/test/util.ts

```diff
@@ -35,20 +35,22 @@ export function readInto<T>(
 	into: T[]
 ): Promise<void> {
 	return new Promise<void>((resolve, reject) => {
-		stream.forEach(
-			function (this: any, value: T): void {
-				expect(this).to.equal(undefined);
-				into.push(value);
-			},
-			function (this: any, err?: Error): void {
-				expect(this).to.equal(undefined);
-				if (err) {
-					reject(err);
-				} else {
-					resolve(undefined);
+		stream
+			.forEach(
+				function (this: any, value: T): void {
+					expect(this).to.equal(undefined);
+					into.push(value);
+				},
+				function (this: any, err?: Error): void {
+					expect(this).to.equal(undefined);
+					if (err) {
+						reject(err);
+					} else {
+						resolve(undefined);
+					}
 				}
-			}
-		);
+			)
+			.catch((e) => reject(e));
```

Comment:
I don't like this change: the code was explicitly created like this to ensure that `readInto()` just captures whatever happened to the arguments of `forEach()`. You can then independently test for `result()` in your test.

If you want the combined behavior (which is probably what most people want that don't write tests like this), just use `stream.toArray()`.